### ISSUE TITLE
Running with reverse proxy

### ISSUE THREAD

**Issue by outofsight** [Nico640/docker-unms#107]
Just started this image, I run behind a reverse proxy, It works but apparently the controller doesn't trust the proxy because, for example, the session tokens table show to the proxy address and not the real client ip address.

There is a way to properly configure the controller to trust the proxy?

**Comment by Nico640** [comment]
Yes, that should be possible. Basically, the nginx config is generated by `/refresh-configuration.sh`, which uses the template located at `/usr/local/openresty/nginx/templates/nginx.conf.template` as the base config.
The relevant part of the template should be this:

```
set_real_ip_from  ${LOCAL_PUBLIC_NETWORK};
real_ip_header    X-Forwarded-For;
real_ip_recursive on;
```

`LOCAL_PUBLIC_NETWORK` is set by `/refresh-configuration.sh` to the network that the docker container is in (e.g. `172.16.0.0/16`)

Is your reverse proxy in the same network as the docker container and does it set the `X-Forwarded-For` header? You can check which network is used by looking at `set_real_ip_from` in `/usr/local/openresty/nginx/conf/nginx.conf`.

**Comment by outofsight** [comment]
In my current setup `uisp-controller` container belogs to two docker networks:

- **basebridge**, a common bridged network that allow the container to access outside world;
- **caddy**, a common overlay network allowing caddy reverse proxy to access the container.

Caddy send the `X-Forwarded-For` header (https://caddyserver.com/docs/caddyfile/directives/reverse_proxy), but `caddy` container access `uisp-controller` from an address of the `caddy` network (10.0...), while `set_real_ip_from` instruction refers to the `basebridge` network (172.20...). So in the login log I see the caddy address (10.0...).

I don't think I can set `LOCAL_PUBLIC_NETWORK` environment variable in docker-compose. It would be nice to have an environment variable to set trusted proxies network list.

I suppose I could do something in docker-compose with a custom entrypoint and `sed` to edit `/usr/local/openresty/nginx/templates/nginx.conf.template` but I'm not so fluent in this.

By the way, I also tried to make `caddy` and `uisp-controller` containers members of the the same `basebridge` network, but this also doens't works. In this case, in the login log, I see the LAN address of the docker host (the same for caddy and uisp-controller), still not the address of my computer.

Any idea?

**Comment by outofsight** [comment]
This can be used to patch the nginx configuration template but it's a bit unmanagebale.

`    entrypoint: bash -c "cp -n /usr/local/openresty/nginx/templates/nginx.conf.template /usr/local/openresty/nginx/templates/nginx.conf.template.bak && sed '/set_real_ip_from\\s*$${LOCAL_PUBLIC_NETWORK};/c\\set_real_ip_from 10.0.0.0/16;\\nset_real_ip_from 192.168.10.16/31;' /usr/local/openresty/nginx/templates/nginx.conf.template.bak > /usr/local/openresty/nginx/templates/nginx.conf.template && cat /usr/local/openresty/nginx/templates/nginx.conf.template | grep set_real_ip_from && /init"
`

**Comment by outofsight** [comment]
> [@outofsight](https://github.com/outofsight) I am not sure of the capabilities of Caddy, but the way I handled it with Traefik was to not expose any ports to the host via Docker (podman) and have Traefik manage everything. From the container, it appeared everything is coming from 80/443/2055 direct.

This is what I do usaully.

But the issue here is to make usip-controller trust the information received by caddy about the client ip address.

Any news @Nico640 ?
